### ISSUE TITLE
Cross Modality Pre-training.

### ISSUE THREAD

**Issue by actionderkokon** [yjxiong/temporal-segment-networks#99]
Hello,in your paper,one strategy for training the ConvNets in temporal segment network is cross modality pre-training whch is applied in  training flow networks, and the accuracy is improved from 81.7% to 86.6%, can you share me how to average the weights across the RGB channels and replicate this average by the channel number of temporal network input before training，in your codes，do u have release this initial method？ thanks for your reply.

**Comment by yjxiong** [comment]
You can see this script for references

https://gist.github.com/yjxiong/de4869c3a5bb3f0f810c

**Comment by actionderkokon** [comment]
Thank u for your reply, i think the meaning u want to convey is : the model which is applied to train temporal network is pre-dealed by averaging the weights across the RGB channels and replicating this average by the channel number of temporal network input ?  so if i directly run the script 'train_tsn.sh',the training accuracy should be around 86.6%,but my accuracy is 81.7% ,this gap seems that the initial method in  your paper is not used in my experience,so i want to know how to make this initial method for the 1st convolutional layer in your paper.

**Comment by yjxiong** [comment]
This script transform a CNN pretrained on ImageNet to an initialization model for the flow modality. This is what "cross-modality" pretraining refers to. So you can use this script to convert your own ImageNet CNNs for temporal, or flow modality.

**Comment by yjxiong** [comment]
The init models we provided have already been processed by this script. If you encounter low accuracy. See this [FAQ](https://github.com/yjxiong/temporal-segment-networks/wiki/Frequently-Asked-Questions#1-caffe-reports-way-lower-accuracies-in-on-the-fly-validation-than-what-you-guys-reported-whats-wrong)

**Comment by actionderkokon** [comment]
It‘s a’ great reply and it is a nice job，i would do these as u have answered. But there is still one thing I can not get  in your reply:"This script transforms a CNN pretrained on ImageNet to an initialization model for the flow modality. "  Do this mean that the script "train_tsn.sh" can transform a still image model to a optical flow model? In other words, which script can  average the weights across the RGB channels and replicate this average by the channel number of temporal network input ? It has confused me for a long time.,

**Comment by yjxiong** [comment]
I meant the script I just linked

https://gist.github.com/yjxiong/de4869c3a5bb3f0f810c

**Comment by actionderkokon** [comment]
Sorry，maybe i can not express the question clearly,  in the codes ,the RGB model used has been processed for initialing the flow network, but if i apply a different network and want to transform a RGB model to a flow model ,how should id do? the script in your link seems no one can achieve this.  Are there any advice?

**Comment by yjxiong** [comment]
You can use the linked script for this purpose.

https://gist.github.com/yjxiong/de4869c3a5bb3f0f810c

That's why I linked it. If you cannot access it, try copy the link address directly to brower.

**Comment by actionderkokon** [comment]
Thanks for your excellent reply.

**Comment by yjxiong** [comment]
Closing this. Please feel free to reopen it if you meet any further problem.

**Comment by actionderkokon** [comment]
ok，I would try some idea，if there are some effective work，i would discuss with u.

**Comment by actionderkokon** [comment]
熊老师您好，我用命令eval_net.py测试ucf101_split1的时候，报如图中的错误，您在平时遇到过么？


zgzhu_xidian@163.com
 
From: yjxiong
Date: 2017-06-01 15:58
To: yjxiong/temporal-segment-networks
CC: actionderkokon; Author
Subject: Re: [yjxiong/temporal-segment-networks] Cross Modality Pre-training. (#99)
Closing this. Please feel free to reopen it if you meet any further problem.

**Comment by ashudeshwal999** [comment]
How to implement this in keras?

**Comment by shabu19** [comment]
can you please provide code for converting resnet101 trained on rgb to optical flow ready model using this script? i am confused in setting the arguments.
